Sources/ApolloCodegenLib/Frontend/JavaScript/src/index.ts

```diff
@@ -63,6 +67,24 @@ export function loadSchemaFromSources(sources: Source[]): GraphQLSchema {
   return schema
 }
 
+function addExperimentalDirectivesToDocument(document: DocumentNode): DocumentNode {
+  // While @defer is experimental the directive needs to be manually added to the
+  // list of available directives for the schema document.
+  return document.definitions.some(definition => 
+    definition.kind == Kind.DIRECTIVE_DEFINITION && 
+    definition.name.value == GraphQLDeferDirective.name
+  ) ?
+    document :
+    concatAST([document, experimentalDeferDirectiveDocumentNode()])
+}
+
+function experimentalDeferDirectiveDocumentNode(): DocumentNode {
+  return {
+    kind: Kind.DOCUMENT,
+    definitions: [definitionNode(GraphQLDeferDirective)]
```

Comment:
There are tests to verify support for the `@defer` directive **and** the arguments it supports because we build the entire `DocumentNode` from the directive definition.